src/js/components/TextInput/TextInput.js

```diff
@@ -435,32 +394,47 @@ const TextInput = forwardRef(
             value={renderLabel(value)}
             readOnly={readOnly}
             onFocus={event => {
-              setFocus(true);
-              if (suggestions && suggestions.length > 0) {
-                announce(messages.suggestionsExist);
-                openDrop();
+              // Don't do anything if we are acting like we already have
+              // focus. This can happen when this input loses focus temporarily
+              // to our drop, see onBlur() handler below.
+              if (!focus) {
+                setFocus(true);
+                if (suggestions && suggestions.length > 0) {
+                  announce(messages.suggestionsExist);
+                  openDrop();
+                }
+                if (onFocus) onFocus(event);
               }
-              if (onFocus) onFocus(event);
             }}
             onBlur={event => {
-              setFocus(false);
-              if (onBlur) onBlur(event);
+              // Only treat it as a blur if the element receiving focus
+              // isn't in our drop. The relatedTarget will be our drop
+              // when the user clicks on a suggestion or interacts with the
+              // scrollbar in the drop.
```

Comment:
helpful comment, thanks. I confirmed the behavior works properly with a long list of suggestions that introduces a scrollbar.